TCAT/Controllers/RouteOptionsViewController.swift

```diff
@@ -194,6 +198,19 @@ class RouteOptionsViewController: UIViewController {
         self.definesPresentationContext = true
         hideSearchBar()
     }
+    
+    private func setUpRouteRefreshing() {
+        let now = Date()
+        let hourMinuteComponents: Set<Calendar.Component> = [.hour, .minute]
+        let nowTime = Calendar.current.dateComponents(hourMinuteComponents, from: now)
+        let lastRefreshTime = Calendar.current.dateComponents(hourMinuteComponents, from: lastRouteRefreshDate)
+        if nowTime != lastRefreshTime {
+            refreshRoutesAndTime()
```

Comment:
Just curious, y do we have to have this call in the initialization?

Reply:
It's called in viewWillAppear so that when somebody goes back to this VC from the route detail VC it only refreshes time and routes if the time changed. I chose to not include this bit of code within refreshRoutesAndTime so that the function produces the expected results in any call regardless of time.